lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -0,0 +1,387 @@
+                    "numVirtualFunctionRequested": 0,
+                    "rebootRequired": false,
+                    "sriovCapable": true
+                }
+            },
+            "changes": {
+                "host_test": {
+                    "numVirtualFunction": 8,
+                    "rebootRequired": true,
+                    "sriovEnabled": true
+                }
+            }
+        }
+    }
+"""
+
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+from time import sleep
+
+
+class VmwareAdapterConfigManager(PyVmomi):
+    """Class to configure SR-IOV settings"""
+
+    def __init__(self, module):
+        super(VmwareAdapterConfigManager, self).__init__(module)
+        cluster_name = self.params.get("cluster_name", None)
+        esxi_host_name = self.params.get("esxi_hostname", None)
+
+        self.vmnic = self.params.get("vmnic", None)
+        self.num_virt_func = self.params.get("num_virt_func", None)
+        self.sriov_on = self.params.get("sriov_on", None)
+
+        #  prepare list of hosts to work with them
+        self.hosts = self.get_all_host_objs(
+            cluster_name=cluster_name, esxi_host_name=esxi_host_name
+        )
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system.")
+        self.results = {"before": {}, "after": {}, "changes": {}}
+
+    def sanitize_params(self, before, hostname, vmnic):
+        """
+        :before     : dict, of params on target interface before changing
+        :hostname   : str, hosthame
+        :vmnic      : srt, target nic
+        :return     : dict, of params forwarded to vCenter, or raise fail in case iconsistancy
+        """
+        params_to_change = {
+            "sriovEnabled": None,
+            "numVirtualFunction": None,
+            "msg": "",
+            "change": False,
+            "changes": {},
+        }
+        # change = False
+        if self.num_virt_func < 0:
+            self.module.fail_json(msg="allowed value for num_virt_func >= 0")
+
+        if self.num_virt_func == 0:
+            if self.sriov_on is True:
+                self.module.fail_json(
+                    msg="with sriov_on == true,  alowed value for num_virt_func > 0"
+                )
+            params_to_change.update({"sriovEnabled": False, "numVirtualFunction": 0})
+
+        if self.num_virt_func > 0:
+            if self.sriov_on is False:
+                self.module.fail_json(
+                    msg="with sriov_on == false,  alowed value for num_virt_func is 0"
+                )
+            params_to_change.update(
+                {"sriovEnabled": True, "numVirtualFunction": self.num_virt_func}
+            )
+
+            # check compatability
+            if not before["sriovCapable"]:
```

Comment:
checking compatibility can be a different method. It doesn't have to be under `sanitize_params`.
Or change the method name to something more appropriate

Reply:
thank you, I'll do separate method.